modules/internal/ChapelDistribution.chpl

```diff
@@ -780,7 +795,11 @@ module ChapelDistribution {
 
     proc ref commit() {
       if cur >= 1 then
-        obj.dsiBulkAddNoPreserveInds(buf[..cur-1]);
+        obj.dsiBulkAddNoPreserveInds(
+          buf[..cur-1], 
+          dataSorted=this.dataSorted, 
```

Comment:
If you are partial towards using `this` explicitly here, I'd recommend `this.buf` above, too. If it was me, I would just pass arguments without naming them or `this`:

`obj.dsiBulkAddNoPreserveInds(buf[..cur-1], dataSorted, isUnique);`

is equally self-documenting IMO.